### PR TITLE
Fully removes departmentType var from Requests Consoles

### PR DESCRIPTION
**Pull request by Profakos**
## About The Pull Request

I was thinking about updating the requests console's UI, when I noticed that #67219 started the process of moving away from the ancient departmentType field. I have decided to put off the UI update until later, and finished the necessary map updates using the power of UpdatePaths. 

Also autodocced the requests console vars and procs.

## Why It's Good For The Game

Finishes the removal of a deprecated (and may I say, badly named) variable. Mappers will not accidentally try to set the outdated bitflag fields, instead they will the more descriptive fields.

## Changelog

Nothing player facing

### CODE REVIEW

**Comment by Profakos** [comment]
Flake test from medibot...

File: code/game/machinery/requests_console.dm [hunk 1]
```diff
@@ -140,15 +156,6 @@ GLOBAL_LIST_EMPTY(req_console_ckey_departments)
...
-	if(departmentType) // Do we have department type flags? Old, deletable once all req consoles are cleaned
-		if((departmentType & REQ_DEP_TYPE_ASSISTANCE) && !(department in GLOB.req_console_assistance))
-			assistance_requestable = TRUE
-
-		if((departmentType & REQ_DEP_TYPE_SUPPLIES) && !(department in GLOB.req_console_supplies))
-			supplies_requestable = TRUE
-
-		if((departmentType & REQ_DEP_TYPE_INFORMATION) && !(department in GLOB.req_console_information))
-			anon_tips_receiver = TRUE
 	// once all request consoles on every map are cleaned, this section above can be deleted
```
Comment (ZephyrTFA) [inline code/game/machinery/requests_console.dm:159]:
```suggestion
```

**Review by ZephyrTFA — CHANGES_REQUESTED** [review]

**Comment by san7890** [comment]
all the brownie points in the world would be yours if you managed to subtype this all somehow so we aren't reliant on having to update maps whenever we update codeside in case we conformationally alter requests consoles. might be out of scope for this PR.

**Comment by Profakos** [comment]
A subtype for each of the seven combinations? I worried a second for the bloat caused by the directional helpers, but they work well for the lamb subtypes, so I might do it in a future PR.

**Comment by san7890** [comment]
> A subtype for each of the seven combinations? I worried a second for the bloat caused by the directional helpers, but they work well for the lamb subtypes, so I might do it in a future PR.

yeah it would be quite bloaty as far as the environment tree is concerned but if we're going to re-use the same exact prefabrication in all five maps it doesn't make sense to not subtype it

**Comment by Profakos** [comment]
Alrighty, I will consider it, but I would indeed want to do this in a future PR.

**Comment by Profakos** [comment]
Looking further into the code in preparation of eventually tgUI-fying this, this could be even simplified further. Assistance_requestable, supplies_requestable and anon_tips_receiver each filter the list of departments, sure, but whatever you pick, it will send the same anonymous message. The three could be replaced by a "Can receive messages" var even.

Though instead I would like to give them an actual functionality in the future, where they add an icon and perhaps a coloured border to the messages, so they are actually different.

**Comment by ShizCalev** [comment]
Why not use mapping helpers instead of subtypes?

**Comment by Profakos** [comment]
I thought about one more helper/subtype if i do that in a future pr, which would be about allowing announcements. Could make it a /head subtype if only head consoles have the ability to make announcements, that is.

**Pull request merged**